input-profile.ts

```diff
@@ -0,0 +1,360 @@
+import {Component, Object3D, Emitter} from '@wonderlandengine/api';
+import {HandTracking} from './hand-tracking.js';
+import {vec3, quat} from 'gl-matrix';
+import {property} from '@wonderlandengine/api/decorators.js';
+import {VrModeActiveSwitch} from './vr-mode-active-switch.js';
+
+const _tempVec = vec3.create();
+const _tempQuat = quat.create();
+const _tempRotation1 = new Float32Array(4);
+const _tempRotation2 = new Float32Array(4);
+const minTemp = new Float32Array(3);
+const maxTemp = new Float32Array(3);
+const hands = ['left', 'right'];
+
+interface VisualResponse {
+    target: Object3D;
+    min: Object3D;
+    max: Object3D;
+    id: number;
+}
+
+/**
+ * Dynamically load and map input profiles for XR controllers.
+ */
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    /**
+     * A cache to store loaded profiles for reuse.
+     */
+    static Cache: Map<string, any> = new Map();
+    private _gamepadObjects: Record<string, Object3D> = {};
+    private _controllerModel: Object3D | null = null;
+    private _defaultControllerComponents: Component[] | undefined;
+    private _handedness!: string;
+    private _profileJSON: any = null;
+    private _buttons: VisualResponse[] = [];
+    private _axes: VisualResponse[] = [];
+
+    /**
+     * The XR gamepad associated with the current input source.
+     */
+    gamepad: Gamepad | undefined;
+    /**
+     * A reference to the emitter which triggered on model lodaed event.
+     */
+    onModelLoaded: Emitter = new Emitter();
+
+    /**
+     * Returns url of input profile json file
+     */
+    url!: string;
+
+    /**
+     * A set of components to filter during component retrieval.
+     */
+    toFilter: Set<string> = new Set(['vr-mode-active-mode-switch']);
+
+    /**
+     * The index representing the handedness of the controller (0 for left, 1 for right).
+     */
+    @property.enum(hands, 0)
+    handednessIndex: number = 0;
+
+    /**
+     * The base path where XR input profiles are stored.
+     */
+    @property.string(
+        'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@latest/dist/profiles/'
+    )
+    path!: string;
+
+    /**
+     * An optional folder path for loading custom XR input profiles.
+     */
+    @property.string()
+    customProfileFolder!: string;
```

Comment:
We should rename that to `basePath` or `baseProfilePath`. That's what it's called in general, something precededing a URL.

Reply:
In that case we might have to make a distinction between this and "path" (https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@latest/dist/profiles/) since they serve different purpose 

path contains basepath which contains ip for all headset's controllers from which headset select the ip based on loaded controller model.

customProfileFolder only contains 1 ip data of a controller independant of the headset in which its loaded 

Will do 

path --> DefaultBasePath
customProfileFolder --> CustomBasePath